docs/source/en/model_doc/wav2vec2.md

```diff
@@ -14,84 +14,150 @@ rendered properly in your Markdown viewer.
+### Model Architecture
+Wav2Vec2 is a speech model that accepts a float array corresponding to the raw waveform of the speech signal. The model was trained using connectionist temporal classification (CTC), so the model output has to be decoded using [`Wav2Vec2CTCTokenizer`].
 
+### Flash Attention 2 Support
+For faster inference, you can use Flash Attention 2. First, install it:
+```bash
+pip install -U flash-attn --no-build-isolation
+```
 
-<div style="text-align: center">
-<img src="https://huggingface.co/datasets/kamilakesbi/transformers_image_doc/resolve/main/data/Wav2Vec2_speedup.png">
-</div>
+Then load the model with Flash Attention 2:
+```python
+from transformers import Wav2Vec2Model
+import torch
+
+model = Wav2Vec2Model.from_pretrained(
+    "facebook/wav2vec2-large-960h-lv60-self", 
+    torch_dtype=torch.float16, 
+    attn_implementation="flash_attention_2"
+).to(device)
+```
 
+### Performance Improvements
+Flash Attention 2 provides significant speedups compared to the native implementation. The expected speedup diagram shows average improvements on the `librispeech_asr` validation split.
 
+### Important Notes
+- The `head_mask` argument is only effective with `attn_implementation="eager"`.
+- Meta (FAIR) released [Wav2Vec2-BERT 2.0](https://huggingface.co/docs/transformers/en/model_doc/wav2vec2-bert) pretrained on 4.5M hours of audio - recommended for fine-tuning tasks.
+- For batch decoding with language models, use [`Wav2Vec2ProcessorWithLM.batch_decode`] with a managed multiprocessing pool for better performance.
+
+### Batch Decoding Example
+```python
+from multiprocessing import get_context
+from transformers import AutoProcessor, AutoModelForCTC
+from datasets import load_dataset
+import torch
+
+model = AutoModelForCTC.from_pretrained("patrickvonplaten/wav2vec2-base-100h-with-lm").to("cuda")
+processor = AutoProcessor.from_pretrained("patrickvonplaten/wav2vec2-base-100h-with-lm")
+
+# Load dataset
+dataset = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation")
+
+def map_to_pred(batch, pool):
+    inputs = processor(batch["speech"], sampling_rate=16_000, padding=True, return_tensors="pt")
+    inputs = {k: v.to("cuda") for k, v in inputs.items()}
+    
+    with torch.no_grad():
+        logits = model(**inputs).logits
+    
+    transcription = processor.batch_decode(logits.cpu().numpy(), pool).text
+    batch["transcription"] = transcription
+    return batch
+
+# Use managed pool for batch decoding
+with get_context("fork").Pool(processes=2) as pool:
+    result = dataset.map(
+        map_to_pred, batched=True, batch_size=2, fn_kwargs={"pool": pool}
+    )
+```
 
 ## Resources
```

Comment:
Remove all the resources except for links to **external** ones (ie, not guides or scripts you can find in the docs)